### ISSUE TITLE
Support specifying type locators for type operator in SpEL

### ISSUE THREAD

**Issue by eyusupov** [thymeleaf/thymeleaf-spring#59]
SpEL allows to register custom TypeLocators for use with T() operator. It would be very convenient if there was an easy way to register custom type locators in Thymeleaf-Spring and be able to write T(Example).method() instead of T(com.example.Example).method()

**Comment by tschuehly** [comment]
Yes, I second this. For my thymeleaf library it would be awesome if I could create TypeLocators at Runtime per ViewComponent!

**Comment by danielfernandez** [comment]
The `TypeLocator` infrastructure in the current versions of Thymelaf is under tight control due to the security restrictions implemented by the restricted execution mode, so this would be no longer feasible. Closing.
